### PR TITLE
TXAD-1: Implement Language, Time Zone and Login API.

### PR DESCRIPTION
**Pull request by Aditi-005**

### CODE REVIEW

File: lib/blocs/login/login_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import 'login_states.dart';
+
+class LoginBloc extends Bloc<LoginEvents, LoginStates> {
+  final LoginRepository _loginRepository = getIt<LoginRepository>();
+  final CustomerCache _customerCache = getIt<CustomerCache>();
+
+  LoginStates get initialState => LoginInitial();
+
+  LoginBloc() : super(LoginInitial()) {
+    on<UserTypeDropDown>(_userTypeValueChanged);
```
Comment (objectorientguy) [inline lib/blocs/login/login_bloc.dart:22]: Rename UserTypeDropdown to UserType

File: lib/blocs/login/login_bloc.dart
Comment (objectorientguy) [inline lib/blocs/login/login_bloc.dart:1]: Check overall naming conventions. For ex :

GenerateOtpLoginLoaded to OTPGenerated and so on...

**Review by objectorientguy — CHANGES_REQUESTED** [review]

File: lib/blocs/login/login_bloc.dart [hunk 2]
```diff
@@ -43,28 +42,27 @@ class LoginBloc extends Bloc<LoginEvents, LoginStates> {
...
-    emit(GenerateOtpLoginLoading());
+  FutureOr<void> _generateLoginOtp(
+      GenerateLoginOtp event, Emitter<LoginStates> emit) async {
+    emit(GeneratingOtpLogin());
     try {
       String? email =
           await _customerCache.getEncryptedEmail(CacheKeys.encryptedEmail);
       Map generateOtpMap = {'emailaddress': email};
-      GenerateOtpLoginModel generateOtpLoginModel =
+      GenerateLoginOtpModel generateOtpLoginModel =
```
Comment (objectorientguy) [inline lib/blocs/login/login_bloc.dart:63]: generateOtpLoginModel ? or generateLoginOtpModel ? 

And anyways you are always going to generate otp for login, so you can say generateOtpModel! Also for rest of the fucntions

File: lib/screens/onboarding/widgets/language_initializing.dart [hunk 3]
```diff
@@ -11,8 +11,8 @@ class LanguageInitializing {
       builder: (context) {
         return Center(
             child: Container(
-                height: 50,
-                width: 220,
+                height: kLanguagePopUpHeight,
```
Comment (objectorientguy) [inline lib/screens/onboarding/widgets/language_initializing.dart:14]: This should be pop up height and not languagePopUpHeight. Make this as a common pop up.

File: lib/screens/onboarding/widgets/select_language_body.dart [hunk 4]
```diff
@@ -34,7 +34,9 @@ class SelectLanguageBody extends StatelessWidget {
                         ));
                   },
                   leading: Padding(
-                      padding: const EdgeInsets.only(top: 8, bottom: 8),
+                      padding: const EdgeInsets.only(
+                          top: kLanguageFlagPadding,
```
Comment (objectorientguy) [inline lib/screens/onboarding/widgets/select_language_body.dart:38]: I dont want padding according to LanguageFlag. It should be generic

File: lib/configs/app_dimensions.dart [hunk 5]
```diff
@@ -22,3 +23,6 @@ const double kLanguageFlagHeight = 35;
 const double kTagsHeight = 25;
 const double kDateTimePickerContainerHeight = 220;
 const double kDateTimePickerHeight = 170;
+const double kLanguagePopUpHeight = 50;
+const double kLanguagePopUpWidth = 220;
```
Comment (objectorientguy) [inline lib/configs/app_dimensions.dart:27]: See if you can keep it generic and not language / tags something like this.

**Review by objectorientguy — CHANGES_REQUESTED** [review]

File: lib/blocs/language/language_bloc.dart [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+      }
+      emit(LanguageKeysLoaded(getLanguageKeysModel: getLanguageKeysModel));
+    } catch (e) {
+      emit(LanguageKeysError(message: e.toString()));
+    }
+  }
+
+  FutureOr<void> _downloadLanguage(
+      DownloadLanguage event, Emitter<LanguageStates> emit) async {
+    emit(DownLoadLanguageLoading());
```
[the remark this answers is not in the document]
Comment (ruchawajpe) [reply lib/blocs/language/language_bloc.dart]: Cool!

**Review by objectorientguy — APPROVED** [review]

File: lib/screens/onboarding/language/select_language_screen.dart [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                      context, StringConstants.kInitializingLanguage);
+                }
+                if (state is LanguageKeysFetched) {
+                  GenericLoadingPopUp.dismiss(context);
+                  Navigator.pushNamed(context, SelectTimeZoneScreen.routeName);
+                }
+                if (state is LanguageKeysError) {
+                  GenericLoadingPopUp.dismiss(context);
+                  showCustomSnackBar(
+                      context, StringConstants.kSelectLanguageAgain, '');
```
Comment (ruchawajpe) [inline lib/screens/onboarding/language/select_language_screen.dart:56]: Add kOk ----> ' '

**Review by ruchawajpe — APPROVED** [review]

**Pull request merged**